### ISSUE TITLE
Latent Assignment

### ISSUE THREAD

**Issue by AjaniBilby** [qupa-lang/Qupa#46]
*Not to be confused with latent writes*
Do not run computations until the value is going to be used.

**Example**:
```qupa
a = 3 + 2;
b = 6 + 4;
print(a);
print(b);
```
```llvm
%1 = add i32 3, 2
call void print(%1)
%2 = add i32 6, 4
call void print(%2)
```

**Benefits**:
  + This can enable latent allocation #19, since the allocation is simply an assignment of an address in IR
  + Unused computations are not executed

**Short-comings**:
  * This will not remove computation such as ``(2*3) + 4``, as ``(2*3)`` is being used in ``? + 4``, even though the final operation may never be used.

**Comment by AjaniBilby** [comment]
**Risk**: Function calls may be delayed due to this behaviour. Make sure any functions which have side effects are executed in order.

**Comment by AjaniBilby** [comment]
This will be resolved by #50
